pkg/apis/core/types_cloudprofile.go

```diff
@@ -73,6 +73,18 @@ func (c *CloudProfile) GetProviderType() string {
 	return c.Spec.Type
 }
 
+// SeedSelector contains constraints for selecting seed to be usable for shoots using a profile
+type SeedSelector struct {
+	// LabelSelector is optional and can be used to select seeds by their label settings
+	// +optional
+	*metav1.LabelSelector
+	// ProviderTypes contains a list of allowed provider types used by the Gardener scheduler to restricting seeds by
```

Comment:
Could you please update `docs/concepts/scheduler.md` as it currently states as limitation
```
- Currently, shoots can only scheduled to seeds from the same cloud provider (`spec.provider.type`), however that is not a technical limitation and might be changed in the future.
```

which is something that this PR aims to improve.